app/src/main/java/org/oppia/app/drawer/NavigationDrawerFragmentPresenter.kt

```diff
@@ -84,16 +86,31 @@ class NavigationDrawerFragmentPresenter @Inject constructor(
 
   private fun openActivityByMenuItemId(menuItemId: Int) {
     if (previousMenuItemId != menuItemId && menuItemId != 0) {
-      val intent = when (NavigationDrawerItem.valueFromNavId(menuItemId)) {
+      when (NavigationDrawerItem.valueFromNavId(menuItemId)) {
         NavigationDrawerItem.HOME -> {
-          Intent(fragment.activity, HomeActivity::class.java)
+          val intent = Intent(fragment.activity, HomeActivity::class.java)
+          fragment.activity!!.startActivity(intent)
+          fragment.activity!!.finish()
         }
         NavigationDrawerItem.HELP -> {
-          Intent(fragment.activity, HelpActivity::class.java)
+          val intent = Intent(fragment.activity, HelpActivity::class.java)
+          fragment.activity!!.startActivity(intent)
+          fragment.activity!!.finish()
+        }
+        NavigationDrawerItem.SWITCH_PROFILE -> {
+          AlertDialog.Builder(fragment.context!!, R.style.AlertDialogTheme)
+            .setMessage(R.string.home_activity_back_dialog_message)
+            .setNegativeButton(R.string.home_activity_back_dialog_cancel) { dialog, _ ->
+              dialog.dismiss()
+            }
+            .setPositiveButton(R.string.home_activity_back_dialog_exit) { _, _ ->
+              // TODO(#322): Need to start intent for ProfileActivity to get update. Change to finish when live data bug is fixed.
+              val intent = Intent(fragment.activity, ProfileActivity::class.java)
+              intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP)
+              fragment.activity!!.startActivity(intent)
```

Comment:
Prefer not to call intent this way. You might have found this in some Profile related code but that will be updated with time.
Instead prefer calling intend similar to HomeActivity code. Mainly you will need to create a companion object for intent in the ProfileActivity and then you can call that function wherever you want.